### ISSUE TITLE
Input data type

### ISSUE THREAD

**Issue by brillkite** [JeremyOng96/A-Hybrid-Method-for-Pavement-Crack-Width-Measurement#4]
Hello, I have been greatly inspired by your paper and found my way to this GitHub page. I would like to try running the measurement_method.py code, but I confront below problem.

ValueError                                Traceback (most recent call last)
<timed exec> in <module>

~\AppData\Local\Temp\ipykernel_4676\1118202690.py in shortest_method(pois, dt, bin_img, S, B, b, verbose, visualize)
    335         B_prime_2 = B_matrix_2 - poi_vector
    336         pt1 = B_matrix_1[np.argmin(np.linalg.norm(B_prime_1,axis=1))]
--> 337         pt2 = B_matrix_2[np.argmin(np.linalg.norm(B_prime_2,axis=1))]
    338 
    339         distance = np.linalg.norm(pt1-pt2)

ValueError: attempt to get argmin of an empty sequence

How can I solve this error?

**Comment by brillkite** [comment]
> _No description provided._

**Comment by JeremyOng96** [comment]
Hi, this error occurs because one of the candidate groups is empty. A candidate group is decided using the orthogonal vector. So if it's empty then it cannot find the shortest distance from the skeleton point to the nearest point. You can probably do a workaround for your case where you identify if there is a empty list then ignore that.

**Comment by brillkite** [comment]
First of all, I had difficulty understanding the code, but thanks for your response. I am just a student working on a project. If I implement exception handling to ignore empty lists, does that mean I should skip the width estimation for that particular image?

**Comment by JeremyOng96** [comment]
Hi, because this measurement is done on every skeleton point and it will produce a dense measurement, I think it's possible to just use an exception to skip that point. However, before doing that, you should take a look whether it will have significant effects on your results.

**Comment by brillkite** [comment]
Yes, I am working with crack images. In that case, it seems possible to handle exceptions and compare the results using if statements or similar methods directly. I will need to check if it will have any impact on the results. Thank you for your response.

**Comment by brillkite** [comment]
I'm sorry, but I have a question about the get_adaptive_region_pca function code while trying to resolve the empty_sequence issue according to your advice. What is the value returned by v when handling edge cases in the above code? I couldn't find a pre-defined v within get_adaptive_region_pca, so I'm not sure what it is...

**Comment by JeremyOng96** [comment]
Hi, I think it is a mistake made by me. Thank you for pointing that out. I will need to update the code a lot, but lately, I am pretty tied up. I can try to make the code work just enough to solve your problem, but I probably need more time to tidy it up.

**Comment by brillkite** [comment]
I'm really grateful for you telling me that. I'd be happy if there are any updates in the future. Thank you!
